language.go

```diff
@@ -116,6 +137,54 @@ func PrefixExtension(r rune, ext func(context.Context, *Parser) (Evaluable, erro
 	return l
 }
 
+// PrefixLanguage switches the parsing language from the current language to the
+// given language. When the subordinate language cannot consume any more input
+// from the parser, control is returned to the current language and the callback
+// is invoked to decide how to continue.
+func PrefixLanguage(r rune, sub Language, ext func(context.Context, *Parser, Evaluable) (Evaluable, error)) Language {
+	l := newLanguage()
+	l.prefixes[r] = func(c context.Context, p *Parser) (Evaluable, error) {
+		eval, err := sub.ChainEvaluable(c, p)
+		if err != nil {
+			return nil, err
+		}
+
+		return ext(c, p, eval)
+	}
+	return l
+}
+
+// Init is a language that does no parsing, but invokes the given function when
+// parsing starts. It is incumbent upon the function to call ParseExpression to
+// continue parsing.
+//
+// This function can be used to customize the parser settings, such as
+// whitespace or ident behavior.
+func Init(ext func(context.Context, *Parser) (Evaluable, error)) Language {
+	l := newLanguage()
+	l.init = ext
+	return l
+}
+
+// Alternate is a language that runs the given function if no other prefix
+// matches.
+func Alternate(ext func(context.Context, *Parser) (Evaluable, error)) Language {
```

Comment:
I'm not sure here. I would have called the methods DefaultExtension. What is your opinion?

Reply:
Sounds good to me!